modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/AbstractExtensionObjectFactory.java

```diff
@@ -182,72 +174,27 @@ private String unwrapChildKey(String key) {
-      throws ConfigurationException {
-    Optional<List<ParameterGroup>> exclusiveGroups =
-        model.getModelProperty(ParameterGroupModelProperty.class).map(mp -> mp.getExclusiveGroups());
-
-    if (exclusiveGroups.isPresent()) {
-      checkParameterGroupExclusiveness(model, exclusiveGroups.get(), resolverKeys);
-    }
-  }
-
-  /**
-   * Checks that the following conditions are honored:
-   * <ul>
-   * <li>Resolved fields from the parameter group belong to the same class</li>
-   * <li>If there is more than one field set, the parameter group declaring those fields must be a nester parameter group</li>
-   * <li>If set, the "one parameter should be present" condition must be honored</li>
-   * </ul>
-   *
-   * @param model
-   * @param exclusiveGroups
-   * @param resolverKeys
-   * @throws ConfigurationException if exclusiveness condition is not honored
-   */
-  private void checkParameterGroupExclusiveness(EnrichableModel model, List<ParameterGroup> exclusiveGroups,
-                                                Set<String> resolverKeys)
+  protected void checkParameterGroupExclusiveness(ParameterizedModel model, Set<String> resolverKeys)
       throws ConfigurationException {
-    for (ParameterGroup<?> group : exclusiveGroups) {
-      Multimap<Class<?>, Field> parametersFromGroup = ArrayListMultimap.create();
-      group.getOptionalParameters().stream()
-          .filter(f -> resolverKeys.contains(f.getName()))
-          .forEach(f -> parametersFromGroup.put(f.getDeclaringClass(), f));
-
-      group.getModelProperty(ParameterGroupModelProperty.class).ifPresent(mp -> mp.getGroups().stream()
-          .flatMap(g -> ((ParameterGroup<Field>) g).getOptionalParameters().stream())
-          .filter(f -> resolverKeys.contains((f).getName()))
-          .forEach(f -> parametersFromGroup.put(f.getDeclaringClass(), f)));
-
-      if (parametersFromGroup.keySet().size() > 1) {
-        throw buildExclusiveParametersException(model, parametersFromGroup);
-      }
 
-      for (Class<?> declaringParameterGroupClass : parametersFromGroup.keySet()) {
-        if (parametersFromGroup.get(declaringParameterGroupClass).size() > 1
-            && declaringParameterGroupClass.equals(group.getType())) {
-          throw buildExclusiveParametersException(model, parametersFromGroup);
+    for (ParameterGroupModel group : model.getParameterGroupModels()) {
+      for (ExclusiveParametersModel exclusiveModel : group.getExclusiveParametersModels()) {
+        Collection<String> definedExclusiveParameters = intersection(exclusiveModel.getExclusiveParameterNames(), resolverKeys);
+        if (definedExclusiveParameters.isEmpty() && exclusiveModel.isOneRequired()) {
+          throw new ConfigurationException((createStaticMessage(format(
+                                                                       "Parameter group '%s' requires that one of its optional parameters should be set but all of them are missing",
```

Comment:
we can communicate the name of the optional parameters from where the user should pick one to provide